### DISCUSSION TITLE
How to build the keyboard?

### DISCUSSION THREAD

**Discussion by mquinson** [betrusted-io/betrusted-hardware#6] [Q&A] [answered]
Hello,

I'm working on another OpenHardware project, and I'd love to integrate a keyboard similar to yours. I was delighted to see that you provide all files, but I must confess that I feel a bit noob here. Do you have any hints on the fabrication process, please? Did you reprap these, or how did you proceed? Any reference to the maker or the material used to make it yourself would be very welcome, please.

Thanks in advance (and kudos for your inspiring project),
Mt

**Comment by bunnie** [comment, marked as answer]
Hey, unfortunately, the answer to this isn't great. There's a tiny shop in the middle of nowhere Dongguang that's sort of hanging on by a thread because it used to make gobs and gobs of keyboards like this back when keyboards like this were a thing. It's now basically just a guy and four helpers in a rented room on the fifth floor of a remote industrial building now, mostly making membrane keyboards for 10-key feature phones shipped to developing markets. To that end, I pre-ordered a *lot* of keyboard membranes, just to protect myself in case the guy goes out of business. I'm not even sure he's around anymore tbh, since the pandemic and all the lockdowns, there is a distinct possibility he's not in business anymore. But...I do have enough stock to build my devices for quite some time.

The basic process is as follows:

1. Create a one-part aluminum mold, with the correct shape and texture for the buttons
2. Pour liquid UV-cure plastic into the mold
3. De-gas the UV-cure liquid in a vacuum chamber
4. Place a clear plastic sheet on the back
5. Cure the UV-cure liquid in a UV irradiator
6. Take the cured assembly and run it through a silkscreening process (as many screens as there are colors)
7. Take the finished assemblies and run them through a CNC cutter to release the keyboards
8. Separately there is a flow that cuts out the button-pusher plastic plate that touches down onto the dome switches. This is bonded onto the keyboard front with adhesives
9. Separately there is a process that places dome switches onto a sticker membrane using an automated machine
10. Quality control & inspection

Of course the keyboards are processed in batch, so a single sheet of plastic has dozens of keyboards, so they can crank out thousands per day with just a few guys running the line. The MOQ is actually really big, which is part of the reason so many languages are offered in the packaging -- the only way I could hit the MOQ was to make multiple variants. The variants are just silkscreen patterns, so, they all counted toward the MOQ.

I don't have the name and contact of the guy who does it -- I worked with AQS to source the vendor, but I couldn't even find his shop if I tried on my own. I have to call up my project manager and book a car if I need to visit the guy, but this was all pre-pandemic. Now, I'm not even so sure he exists anymore.

I haven't tried to look, scared to even ask. I've got 99 problems but a lack of keyboard stock ain't one.

**Comment by dumblob** [comment]
In case the whole manufacturing process scares you (as it scares me), feel free to buy newly produced (**not** refurbished) original Blackberry Q10 keyboards (of course, no security guarantees as in case of *betrusted*). I have ordered several from aliexpress from the manufacturer "HKFASTEL" and am really happy with them. I suspect it might even be **the** supplier for Blackberry back then when Q10 was being produced.

The only issue is that the connectors seem pretty much EOL and kind of disappearing from stocks around the world (we talk about higher hundreds to small thousands). I am considering contacting HKFASTEL and asking if they could use some newer connector to sustain the (already limited or non-existent) production.

Feel free to look at [Fairberry](https://github.com/Dakkaron/Fairberry/blob/main/Documentation/Hardware_Fairberry_Mainboard.md ) which is the project I need the Q10 keyboards for :wink:.

**Comment by mquinson** [comment]
Thanks a lot for your answers, both of you. That's really inspiring.
